File: app-mvc-java/src/main/java/com/naveed/samples/data/network/MoviesServiceFactory.java
package com.naveed.samples.data.network;

import com.google.gson.Gson;
import com.google.gson.GsonBuilder;
import java.util.concurrent.TimeUnit;
import com.naveed.samples.helper.utils.AppConstants;
import okhttp3.OkHttpClient;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;


public class MoviesServiceFactory {

    static IMoviesApi iMoviesApi = null;
    public static IMoviesApi getInstance() {

        if (iMoviesApi == null) {
                Gson gson = new GsonBuilder()
                        .setLenient()
                        .create();

                OkHttpClient okHttpClient = new OkHttpClient.Builder().readTimeout(30, TimeUnit.SECONDS)
                        .connectTimeout(30, TimeUnit.SECONDS)
                        .build();

                Retrofit retrofit = new Retrofit.Builder()
                        .baseUrl(AppConstants.BASE_URL)
                        .client(okHttpClient)
                        .addConverterFactory(GsonConverterFactory.create(gson))
                        .build();

            iMoviesApi = retrofit.create(IMoviesApi.class);
            }
            return iMoviesApi;
        }




}

File: app-mvp/src/main/java/com/naveed/samples/data/network/MoviesServiceFactory.kt
package com.naveed.samples.data.network

import com.google.gson.Gson
import com.google.gson.GsonBuilder
import java.util.concurrent.TimeUnit
import com.naveed.samples.helper.utils.AppConstants
import okhttp3.OkHttpClient
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory


object MoviesServiceFactory {

    internal var iMoviesApi: IMoviesApi? = null
    val instance: IMoviesApi
        get() {
            if (iMoviesApi == null) {
                val gson = GsonBuilder()
                        .setLenient()
                        .create()

                val okHttpClient = OkHttpClient.Builder().readTimeout(30, TimeUnit.SECONDS)
                        .connectTimeout(30, TimeUnit.SECONDS)
                        .build()

                val retrofit = Retrofit.Builder()
                        .baseUrl(AppConstants.BASE_URL)
                        .client(okHttpClient)
                        .addConverterFactory(GsonConverterFactory.create(gson))
                        .build()

                iMoviesApi = retrofit.create(IMoviesApi::class.java)
            }
            return iMoviesApi!!
        }


}
